File: app/src/main/java/com/example/fruitsdiary/usecase/addeditentry/selectfruit/SelectFruitContract.java
package com.example.fruitsdiary.usecase.addeditentry.selectfruit;

import com.example.fruitsdiary.BasePresenter;
import com.example.fruitsdiary.BaseView;
import com.example.fruitsdiary.model.Entry;
import com.example.fruitsdiary.model.Fruit;

import java.util.List;

public interface SelectFruitContract {

    interface View extends BaseView {
        void showFruitList(List<Fruit> fruitList);
    }

    interface Presenter extends BasePresenter<View> {
        void populateFruitList();
    }
}

File: app/src/main/java/com/example/fruitsdiary/usecase/addeditentry/selectfruit/SelectFruitContract.kt
package com.example.fruitsdiary.usecase.addeditentry.selectfruit

import com.example.fruitsdiary.BasePresenter
import com.example.fruitsdiary.BaseView
import com.example.fruitsdiary.model.Fruit

interface SelectFruitContract {

    interface View : BaseView {
        fun showFruitList(fruitList: List<Fruit>)
    }

    interface Presenter : BasePresenter<View> {
        fun populateFruitList()
    }
}
